### ISSUE TITLE
The Gradebook MFE makes undue calls to analytics services

### ISSUE THREAD

**Issue by BbrSofiane** [openedx/build-test-release-wg#6]
![mfe analytics](https://user-images.githubusercontent.com/13796098/107238686-0af10f00-6a20-11eb-89bf-e8e645dc742e.png)

I really don’t want my adblocker to report calls to segment or newrelic when I test an MFE without credentials for these services.

**Comment by regisb** [comment]
As far as I know, this issue also affects the Account and Profile MFEs -- and actually all MFEs, since they all depend on frontend-platform, which makes calls to Segment: https://github.com/edx/frontend-platform#service-implementations

This would mean that Lilac is affected by this issue.

I have no idea how to address this...

**Comment by BbrSofiane** [comment]
@sarina @davidjoy any thoughts/suggestion on this?

**Comment by sarina** [comment]
Hrmmm. @davidjoy will know the most about frontend-platform so I mostly defer to him. Seems like these calls should be gated by whether or not you have the proper keys set up.

**Comment by davidjoy** [comment]
Ah, this needs to be addressed in frontend-platform.  If there's no SEGMENT_KEY configured, then it should avoid loading the segment script.  Unfortunately today it just loads it with a `null` as part of the URL where the key would go.

**Comment by BbrSofiane** [comment]
@davidjoy can this be done in time for Lilac?

**Comment by davidjoy** [comment]
I think it's fairly simple to fix, I've created this task for it: https://openedx.atlassian.net/browse/CENG-53

FWIW it doesn't actually create any problems beyond an unexpected 404 response from segment.  I wouldn't hold up the release for it, though when it's fixed it'd be nice to cherry pick it to lilac.

**Comment by davidjoy** [comment]
Okay, so I've fixed this in frontend-app-account.  Turns out that frontend-app-gradebook is a more complicated situation, as it's using its own custom implementation of Segment and doesn't rely on frontend-platform for it.  

I've made a PR there and the owning team will need to review it.  I'm hoping I can cherry-pick it onto lilac.master without incident.

https://github.com/edx/frontend-app-gradebook/pull/181
